cefbrowser/cefbrowser.py

```diff
@@ -798,145 +925,231 @@ def on_paste(self, *largs):
         print("PASTE", t)
 
 
-class ClientHandler():
+class ClientHandler:
     browser_widgets = {}
     pending_popups = {}
 
     def __init__(self, *largs):
         self.browser_widgets = {}
 
-    # DisplayHandler TODO: OnContentsSizeChange, OnFaviconURLChange, OnNavStateChange
-
-    def OnLoadingStateChange(self, browser, is_loading, can_go_back, can_go_forward):
-        bw = self.browser_widgets[browser]
-        bw.is_loading = is_loading
-        bw.can_go_back = can_go_back
-        bw.can_go_forward = can_go_forward
-        if not is_loading:
-            bw.js._inject()
+    # DisplayHandler TODO: OnContentsSizeChange, OnFaviconURLChange,
+    # OnNavStateChange
 
-    def OnAddressChange(self, browser, frame, url):
-        if browser.GetMainFrame()==frame:
+    def OnAddressChange(self, browser, frame, url):  # noqa: N802
+        if browser.GetMainFrame() == frame:
```

Comment:
This could be written: `frame.IsMain()`.

Also do not compare browser/frame objects using `==` operator, because there are some edge cases when this doesn't work. Always use GetIdentifier() on these objects.